cfme/intelligence/chargeback/rates.py

```diff
@@ -235,19 +230,72 @@ def delete(self, cancel=False):
         """
         view = navigate_to(self, 'Details')
         view.toolbar.configuration.item_select('Remove from the VMDB', handle_alert=(not cancel))
-        view = self.create_view(navigator.get_class(self, 'All').VIEW)
-        assert view.is_displayed
+        view = self.create_view(navigator.get_class(self.parent, 'All').VIEW, wait=10)
+        view.flash.assert_no_error()
+
+
+@attr.s
+class ComputeRateCollection(BaseCollection):
+    ENTITY = ComputeRate
+    RATE_TYPE = ENTITY.RATE_TYPE
+
+    def create(self, description, currency=None, fields=None):
+        """ Create a rate in the UI
+
+            Args:
+                description (str): name of the compute rate to create
+                currency (str): - type of currency for the rate
+                fields (dict): -  nested dictionary listing the Rate Details
+                    Key => Rate Details Description
+                    Value => dict
+                        Key => Rate Details table column names
+                        Value => Value to input in the table
+        """
+        rate = self.instantiate(description, currency, fields)
+
+        view = navigate_to(self, 'Add')
+        view.fill_with({'description': rate.description,
+                        'currency': rate.currency,
+                        'fields': rate.fields},
+                       on_change=view.add_button,
+                       no_change=view.cancel_button)
+
+        view = self.create_view(navigator.get_class(self, 'All').VIEW, wait=10)
         view.flash.assert_no_error()
 
+        return rate
+
 
+@attr.s
 class StorageRate(ComputeRate):
     # Methods and form for this are similar to that of ComputeRate, but navigation is different
     # from that of ComputeRate.
     pretty_attrs = ['description']
     RATE_TYPE = 'Storage'
 
 
-@navigator.register(ComputeRate, 'All')
+class StorageRateCollection(BaseCollection):
+    ENTITY = StorageRate
+    RATE_TYPE = ENTITY.RATE_TYPE
+
+    def create(self, description, currency=None, fields=None):
+        # Create a rate in UI
```

Comment:
No doc block for `create` here? Should be the exact same as `ComputeRateCollection.create`